From 1f8ef59665e8eb1c2014488d1046394ce88acdd0 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 30 Jul 2020 15:47:55 +0530
Subject: [PATCH 1/5] octeontx2-af: Check the msix offset return value

commit a5fd8dfa0c8eaf9d12a0097f4a1f8ba5b889ce67 from
git@git.assembla.com:cavium/WindRiver.linux.git

rvu_get_msix_offset can return invalid MSIX vector
which is 0xFFFF. Hence the caller should check the
return value otherwise possible array out of
bounds can occur in rvu_clear_msix_offset function.

Change-Id: I0757a0308d2e4418b46860e8078a73a72cea79ae
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/33195
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 585b9f21d072..69f6fa3eeaf5 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -1828,6 +1828,8 @@ static void rvu_clear_msix_offset(struct rvu *rvu, struct rvu_pfvf *pfvf,
 		    (lf << block->lfshift), cfg & ~0x7FFULL);
 
 	offset = rvu_get_msix_offset(rvu, pfvf, block->addr, lf);
+	if (offset == MSIX_VECTOR_INVALID)
+		return;
 
 	/* Update the mapping */
 	for (vec = 0; vec < nvecs; vec++)
-- 
2.17.1

